### PR TITLE
[ameba] Ameba SNTP and observer support

### PR DESCRIPTION
**Pull request by pankore**
* SNTP Support for src/platform/Ameba/SystemTimeSupport.cpp
* Add Ameba Observer for air-purifier-app, all-clusters-minimal-app, chef-app, light-switch-app, lighting-app, and ota-requestor-app examples

### CODE REVIEW

**Comment by pankore** [comment]
Waiting #36383 to be merged first

File: examples/lighting-app/ameba/chip_main.cmake [hunk 1]
```diff
@@ -192,6 +192,7 @@ target_include_directories(
     ${chip_dir}/zzz_generated/lighting-app
     ${chip_dir}/zzz_generated/lighting-app/zap-generated
     ${chip_dir}/zzz_generated/app-common
+    ${chip_dir}/examples/all-clusters-app/ameba/main/include #to get AmebaObserver.h
```
Comment (andy31415) [inline examples/lighting-app/ameba/chip_main.cmake:195]: having `examples/lighting-app` point to `examples-all-clusters-app` does not look right. Please figure out another way to layer things that makes sense. Several platforms have common or maybe some ameba/observer would be appropriate

Comment (pankore) [reply examples/lighting-app/ameba/chip_main.cmake]: Updated in the last commit, please re-review.

**Review by andy31415 — APPROVED** [review]

File: src/platform/Ameba/SystemTimeSupport.cpp [hunk 2]
```diff
@@ -58,26 +58,32 @@ Milliseconds64 ClockImpl::GetMonotonicMilliseconds64(void)
 
 CHIP_ERROR ClockImpl::GetClock_RealTime(Clock::Microseconds64 & curTime)
 {
-    // TODO(19081): This platform does not properly error out if wall clock has
-    //              not been set.  For now, short circuit this.
-    return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
-#if 0
-    time_t seconds;
-    struct rtkTimeVal tv;
+#if defined(CONFIG_ENABLE_AMEBA_SNTP) && (CONFIG_ENABLE_AMEBA_SNTP == 1)
```
Comment (andy31415) [inline src/platform/Ameba/SystemTimeSupport.cpp:61]: same here: this should be `#if CONFIG_ENABLE_AMEBA_SNTP` with a proper include that guarantees this define is available or not.

File: src/platform/Ameba/SystemTimeSupport.cpp [hunk 3]
```diff
@@ -105,9 +111,16 @@ CHIP_ERROR InitClock_RealTime()
         Clock::Microseconds64((static_cast<uint64_t>(CHIP_SYSTEM_CONFIG_VALID_REAL_TIME_THRESHOLD) * UINT64_C(1000000)));
     // Use CHIP_SYSTEM_CONFIG_VALID_REAL_TIME_THRESHOLD as the initial value of RealTime.
     // Then the RealTime obtained from GetClock_RealTime will be always valid.
-    //
-    // TODO(19081): This is broken because it causes the platform to report
-    //              that it does have wall clock time when it actually doesn't.
+#if defined(CONFIG_ENABLE_AMEBA_SNTP) && (CONFIG_ENABLE_AMEBA_SNTP == 1)
```
[the remark this answers is not in the document]
Comment (pankore) [reply src/platform/Ameba/SystemTimeSupport.cpp]: Yes, it is actually already defined in config/ameba/chip.cmake. Updated in the last commit.

File: src/platform/Ameba/SystemTimeSupport.cpp (hunk 2 above)
Comment (pankore) [reply src/platform/Ameba/SystemTimeSupport.cpp]: Yes, it is actually already defined in config/ameba/chip.cmake. Updated in the last commit.

**Pull request merged**